lsql/judge/tests/test_views.py

```diff
@@ -455,3 +455,39 @@ def test_select_no_output(self):
             response = client.post(submit_url_select, {'code': stmt}, follow=True)
             self.assertTrue(response.json()['veredict'] == VeredictCode.WA)
             self.assertIn('Generado por tu código SQL: 0 columnas', response.json()['feedback'])
+
+    def test_download_submission(self):
+        "Test to download code of submission"
+        client = Client()
+        user = create_user('2222')
+        login_redirect_url = reverse('judge:login')
+        collection = create_collection('Colleccion de prueba TTT')
+        problem = create_select_problem(collection, 'SelectProblem ABC DEF')
+        submission = create_submission(problem, user, VeredictCode.AC, 'select *** from *** where *** and more')
+
+        # Submission redirects to login
+        submission_url = reverse('judge:submission', args=[submission.pk])
+        login_redirect_submission = f'{login_redirect_url}?next={submission_url}'
+        response = client.get(submission_url, follow=True)
+        self.assertEqual(response.redirect_chain,
+                         [(login_redirect_submission, 302)])
+
+        # # Download your own code submission
+        user = create_user('2222', 'tamara')
+        create_user('3333', 'juan')
+        client.login(username='tamara', password='2222')
+        collection = create_collection('Colleccion de prueba TTT')
+        problem = create_select_problem(collection, 'SelectProblem ABC DEF')
+        submission = create_submission(problem, user, VeredictCode.AC, 'select *** from *** where *** and more')
+
+        if submission.user == user:
+            url = reverse('judge:download_submission', args=[submission.pk])
+            response = client.get(url, follow=True)
+            self.assertEqual(
+                response.get('Content-Disposition'),
```

Comment:
Además de comprobar el *Content-Disposition* debes comprobar también que el contenido del fichero descargado es exactamente igual al texto del envío `'select *** from *** where *** and more'`

Reply:
Para comprobarlo, puedo comparar el código del envío con  `str(response.content))` ?

Reply:
Sí, con `str(response.content)` tienes la contestación que genera el juez como cadena de texto. Pero mejor echa un vistazo a los test de unidad que hizo @iburgoa13 para descargar scripts de creación de BD: https://github.com/emartinm/lsql/blob/d4a0f8bbf3d4578e134f85607116187dd49968e5/lsql/judge/tests/test_views.py#L368

Reply:
Como `response.content` es de tipo `bytes`, creo que `str(response.content)` es exactamente lo mismo que `response.content.decode('UTF-8')` pero compruébalo por si acaso

Reply:
Si, he visto que el `response.content` te lo da en bytes y para decodificarlo hay que usar el `response.content.decode` y asi tener una cadena.